Using_str.c

```diff
@@ -0,0 +1,44 @@
+#include <stdio.h>
+#include <string.h>
+
+static char *s_gets(char *string, int number)
+{
+   char *ret_value;
+   int index = 0;
+   ret_value = fgets(string, number, stdin);
+   if(ret_value < 0)
+   {
+      while(string[index] != '\n' && string[index] != '\0')
+      {
+         index++;
+         if(string[index] == '\n')
+         {
+            string[index] = '\0';
+         }
+         else
+            while(getchar() != '\n')
+            {
+               continue;
+            }
+      }
+   }
+   return ret_value;
+}
+
+int main(void)
+{
+   char flower[80];
+   char AddString[] = "s smell like old shoes.";
+   puts("what is your favorite flower?");
+   if(s_gets(flower, 80))
+   {
+      strcat(flower, AddString);
```

Comment:
What will happen if someone enters a flower name that has close to 80 characters in it? Will there be enough space?